From a78961fb02542e6b8c8bb4f99f5816c82446d3ea Mon Sep 17 00:00:00 2001
From: Jan Nieuwenhuizen <janneke@gnu.org>
Date: Mon, 30 Mar 2009 12:40:45 +0200
Subject: [PATCH] sc/source/ui/unoobj/tokenuno.cxx: ConvertToTokenArray must return 0 upon fail.

ScTokenConversion::ConvertTo* traditionally all return nonzero when
successful, which is what callers expect.  This changed for
ConvertToTokenArray when it outsourced parsing to formula/, returning
what rTokenArray.Fill () returns -- which is 0 upon success.
---
 sc/source/ui/unoobj/tokenuno.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/sc/source/ui/unoobj/tokenuno.cxx b/sc/source/ui/unoobj/tokenuno.cxx
index d2ce461..0c76cdb 100644
--- sc/source/ui/unoobj/tokenuno.cxx
+++ sc/source/ui/unoobj/tokenuno.cxx
@@ -345,7 +345,7 @@ void lcl_SingleRefToApi( sheet::SingleReference& rAPI, const ScSingleRefData& rR
 bool ScTokenConversion::ConvertToTokenArray( ScDocument& rDoc,
         ScTokenArray& rTokenArray, const uno::Sequence<sheet::FormulaToken>& rSequence )
 {
-    return rTokenArray.Fill(rSequence,rDoc.GetExternalRefManager());
+    return !rTokenArray.Fill(rSequence,rDoc.GetExternalRefManager());
 }
 
 // static
-- 
1.6.0.rc1.49.g98a8

